### PR TITLE
feat: Project covered by language toggle and documentation

### PR DESCRIPTION
**Pull request by gustavop-dev**
# Description

This Pull Request includes the following changes:

1. **Full Documentation Coverage**:
   - Complete documentation was added for all components, stores, utilities, and composables in the frontend project.
   - Each function, prop, and logic implementation is now documented, following best practices to ensure clarity and maintainability.

2. **Language Toggle Implementation**:
   - A language toggle has been implemented throughout the entire frontend project.
   - The user can now switch between supported languages (e.g., English and Spanish) dynamically.
   - All text content, including navigation links, buttons, and labels, now reflects the selected language, ensuring a consistent multilingual experience.
   - The global language store was updated to manage language changes across views and components efficiently.

# Changes
- Documented components, composables, and utilities.
- Implemented language toggle functionality.
- Updated the language store to handle dynamic text changes across the entire application.

# Additional Information
- No breaking changes introduced.
- The language toggle supports both English and Spanish, and additional languages can be easily integrated if needed.

### CODE REVIEW

**Comment by gustavop-dev** [comment]
In prices.vue and hosting.vue belonging to views in frontend, you won't find documentation because these files are subject to changes.

**Review by carlos18bp — CHANGES_REQUESTED** [review]
frontend/src/views/hosting.vue
1. El nombre del componente deberia empezar con mayuscula
2. Linea 87,88,89,90 no terminan en ;
3. Agrega un espacio entre la linea 82 y 83
4. Barrer el template y cambiar las lineas demasiado largas

frontend/src/views/Prices.vue
1. Agrega un espacio entre la linea 408 y 409

frontend/src/views/Home.vue
1. Linea 71 no termina en ;

frontend/src/views/AboutUs.vue
1. Barrer el template y cambiar las lineas demasiado largas
2. Agrega un espacio entre la linea 76 y 77

frontend/src/views/3dAnimations/List.vue
1. Linea 51 no terminan en ;

frontend/src/components/Home/InitialVideo.vue
1. Barrer el template y cambiar las lineas demasiado largas

frontend/src/components/Layouts/Contact.vue
1. Barrer el template y cambiar las lineas demasiado largas

frontend/src/components/Layouts/Footer.vue
1. Barrer el template y cambiar las lineas demasiado largas
2. Extra line in line 110

frontend/src/components/Layouts/Navbar.vue
1. Barrer el template y cambiar las lineas demasiado largas

frontend/src/components/Utils/ButtonWithArrow.vue
1. En la linea 16 cuando uses querySelector la clase debe inicar con js-{name-clase}, 
Esto con el fin de saber que no es una clase css sino js

frontend/src/components/Utils/SocialLinks.vue
1. En la linea 26 cuando uses querySelector la clase debe inicar con js-{name-clase}, 
Esto con el fin de saber que no es una clase css sino js

frontend/src/components/spline/Backgrounds/Dune.vue
1. Arreglar la sangria

projectapp/content/__pycache__/serializers.cpython-312.pyc
projectapp/content/__pycache__/views.cpython-312.pyc
projectapp/content/models/__pycache__/__init__.cpython-312.pyc
projectapp/content/models/__pycache__/components.cpython-312.pyc
projectapp/content/models/__pycache__/contact.cpython-312.pyc
projectapp/content/models/__pycache__/designs.cpython-312.pyc
projectapp/content/models/__pycache__/models_3d.cpython-312.pyc
1. Esto no deberia estar en el repositorio, ni en tu pr, asegurate que tu .gitignore este bien configurado


Comentarios generales:
1. Todos los comentarios anteriores son de la rama main
2. Seria bueno agregar algunos comentarios dentro de la logica de los template, para saber que se esta haciendo
3. Los metodos no se encuentran documentados
4. En los serializadores de Django, si deseas incluir todos los campos del modelo en tu serializador, puedes utilizar la opción '__all__' en el atributo fields.
4. Una vez se atiendan los comentarios y todo este en la rama main, favor listarme las tareas que se me asignaran

File: frontend/src/components/Layouts/Contact.vue [hunk 1]
```diff
@@ -6,21 +6,42 @@
...
+                      <a 
+                        @click="showModalEmail = true" 
+                        ref="emailLink" 
+                        @mouseover="animateLink" 
+                        @mouseleave="resetLink" 
+                        class="inline-block text-xl absolute bottom-0 text-esmerald cursor-pointer"
+                        >
+                        {{ globalMessages.email_address }}
+                        <span ref="underline" class="absolute left-0 bottom-0 h-0.5 w-0 bg-black transition-all duration-300"></span>
+                        <span ref="arrow" class="absolute right-0 top-1/2 transform -translate-y-1/2 opacity-0 transition-opacity duration-300">➜</span>
```
Comment (carlos18bp) [inline frontend/src/components/Layouts/Contact.vue]: Line too large

File: frontend/src/components/Layouts/Contact.vue [hunk 2]
```diff
@@ -6,21 +6,42 @@
...
+                      </p>
+                      <a 
+                        @click="showModalEmail = true" 
+                        ref="emailLink" 
+                        @mouseover="animateLink" 
+                        @mouseleave="resetLink" 
+                        class="inline-block text-xl absolute bottom-0 text-esmerald cursor-pointer"
+                        >
+                        {{ globalMessages.email_address }}
+                        <span ref="underline" class="absolute left-0 bottom-0 h-0.5 w-0 bg-black transition-all duration-300"></span>
```
Comment (carlos18bp) [inline frontend/src/components/Layouts/Contact.vue]: Line too large

File: frontend/src/components/Layouts/Contact.vue [hunk 3]
```diff
@@ -6,21 +6,42 @@
...
+                      <a 
+                        href="https://wa.me/message/XX77FJEUEM26H1?src=qr" 
+                        target="_blank" 
+                        ref="chatLink" 
+                        @mouseover="animateLinkChat" 
+                        @mouseleave="resetLinkChat" 
+                        class="inline-block text-xl absolute bottom-0 text-esmerald cursor-pointer"
+                        >
+                        {{ globalMessages.chat }}
+                        <span ref="underlineChat" class="absolute left-0 bottom-0 h-0.5 w-0 bg-black transition-all duration-300"></span>
```
Comment (carlos18bp) [inline frontend/src/components/Layouts/Contact.vue]: Line too large

File: frontend/src/components/Layouts/Contact.vue [hunk 4]
```diff
@@ -6,21 +6,42 @@
...
+                        <span ref="arrowChat" class="absolute right-0 top-1/2 transform -translate-y-1/2 opacity-0 transition-opacity duration-300">➜</span>
```
Comment (carlos18bp) [inline frontend/src/components/Layouts/Contact.vue]: Line too large

File: projectapp/content/serializers.py [hunk 5]
```diff
@@ -13,7 +13,7 @@ class ContactSerializer(serializers.ModelSerializer):
     """
     class Meta:
         model = Contact
-        fields = ['email', 'subject', 'message']
+        fields = "__all__"
```
Comment (carlos18bp) [inline projectapp/content/serializers.py]: Use single quotes here: `'__all__'`

File: projectapp/content/serializers.py [hunk 6]
```diff
@@ -33,7 +33,7 @@ class Models3DSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = Models3D
-        fields = ['id', 'title_en', 'title_es', 'image_url', 'file_url']
+        fields = "__all__"
```
Comment (carlos18bp) [inline projectapp/content/serializers.py]: Use single quotes here: `'__all__'`

File: projectapp/content/serializers.py [hunk 7]
```diff
@@ -71,7 +71,7 @@ class DesignsSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = Designs
-        fields = ['id', 'title_en', 'title_es', 'presentation_image_url', 'detail_image_url']
+        fields = "__all__"
```
Comment (carlos18bp) [inline projectapp/content/serializers.py]: Use single quotes here: `'__all__'`

File: projectapp/content/serializers.py [hunk 8]
```diff
@@ -104,7 +104,7 @@ class ExampleSerializer(serializers.ModelSerializer):
     """
     class Meta:
         model = Example
-        fields = ['id', 'title_en', 'title_es', 'image']
+        fields = "__all__"
```
Comment (carlos18bp) [inline projectapp/content/serializers.py]: Use single quotes here: `'__all__'`

File: projectapp/content/serializers.py [hunk 9]
```diff
@@ -124,7 +124,7 @@ class ComponentsSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = Components
-        fields = ['id', 'title_en', 'title_es', 'image_url', 'examples']
+        fields = "__all__"
```
Comment (carlos18bp) [inline projectapp/content/serializers.py]: Use single quotes here: `'__all__'`

File: projectapp/content/serializers.py [hunk 10]
```diff
@@ -151,7 +151,7 @@ class SectionSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = Section
-        fields = ['id', 'title_en', 'title_es', 'components']
+        fields = "__all__"
```
Comment (carlos18bp) [inline projectapp/content/serializers.py]: Use single quotes here: `'__all__'`

File: projectapp/content/serializers.py [hunk 11]
```diff
@@ -171,4 +171,4 @@ class CategoriesDevelopmentSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = CategoriesDevelopment
-        fields = ['id', 'title_en', 'title_es', 'description_en', 'description_es', 'sections']
+        fields = "__all__"
```
Comment (carlos18bp) [inline projectapp/content/serializers.py]: Use single quotes here: `'__all__'`

**Review by carlos18bp — CHANGES_REQUESTED** [review]

File: projectapp/content/serializers.py (hunk 11 above)
Comment (gustavop-dev) [reply projectapp/content/serializers.py]: Done!

File: projectapp/content/serializers.py (hunk 9 above)
Comment (gustavop-dev) [reply projectapp/content/serializers.py]: Done

File: projectapp/content/serializers.py (hunk 8 above)
Comment (gustavop-dev) [reply projectapp/content/serializers.py]: Done

File: projectapp/content/serializers.py (hunk 7 above)
Comment (gustavop-dev) [reply projectapp/content/serializers.py]: Done

File: projectapp/content/serializers.py (hunk 6 above)
Comment (gustavop-dev) [reply projectapp/content/serializers.py]: Done!

File: projectapp/content/serializers.py (hunk 5 above)
Comment (gustavop-dev) [reply projectapp/content/serializers.py]: Done

File: frontend/src/components/Layouts/Contact.vue (hunk 1 above)
Comment (gustavop-dev) [reply frontend/src/components/Layouts/Contact.vue]: Done

File: frontend/src/components/Layouts/Contact.vue (hunk 2 above)
Comment (gustavop-dev) [reply frontend/src/components/Layouts/Contact.vue]: Done

File: frontend/src/components/Layouts/Contact.vue (hunk 3 above)
Comment (gustavop-dev) [reply frontend/src/components/Layouts/Contact.vue]: Done

File: frontend/src/components/Layouts/Contact.vue (hunk 4 above)
Comment (gustavop-dev) [reply frontend/src/components/Layouts/Contact.vue]: Done

**Review by carlos18bp — APPROVED** [review]

**Pull request merged**